src/main/java/edu/umich/its/cpm/BoxUtils.java

```diff
@@ -0,0 +1,189 @@
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Date;
+import java.text.SimpleDateFormat;
+import java.io.PrintWriter;
+import java.io.IOException;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.core.env.Environment;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.client.RestClientException;
+
+import org.json.JSONObject;
+import org.json.JSONArray;
+
+import com.box.sdk.BoxAPIConnection;
+import com.box.sdk.BoxFolder;
+import com.box.sdk.BoxFile;
+import com.box.sdk.BoxItem;
+import com.box.sdk.BoxItem.Info;
+import com.box.sdk.BoxUser;
+
+/**
+ * provides util functions for accessing Box APIs
+ * 
+ * @author zqian
+ *
+ */
+public class BoxUtils {
+
+	private static final int MAX_DEPTH = 0;
+	
+	private static final SimpleDateFormat date_formatter = new SimpleDateFormat(
+			"yyyy-MM-dd-hh.mm.ss");
+	
+	private static final Logger log = LoggerFactory.getLogger(BoxUtils.class);
+
+	public static void authenticate(String boxAPIUrl, String boxClientId,
+			String boxClientRedirectUri, String remoteUserEmail,
+			HttpServletResponse response) {
+		// Box authorization
+		RestTemplate restTemplate = new RestTemplate();
+		
+		if (boxAPIUrl == null)
+		{
+			// log error
+			log.error("No Box API url specified. ");
+			return;
+		}
+		
+		String requestUrl = boxAPIUrl + "/oauth2/authorize"
+				+ "?response_type=code" + "&client_id=" + boxClientId
+				+ "&redirect_uri=" + boxClientRedirectUri
+				+ "&state=&box_login=" + remoteUserEmail;
+		log.info(requestUrl);
+
+		try {
+			String resultString = restTemplate.getForObject(requestUrl,
+					String.class);
+
+			// open window with resultString
+			response.setContentType("text/html");
+			response.getWriter().println(resultString);
+			response.flushBuffer();
+		} catch (RestClientException e) {
+			log.error(requestUrl + e.getMessage());
+		} catch (IOException e) {
+			log.error(requestUrl + e.getMessage());
+		}
+	}
+
+	/**
+	 * get the authCode as embedded from Box callback response
+	 */
+	public static String getAuthCodeFromBoxCallback(HttpServletRequest request) {
+		// get the code String from Box authorization callback
+		String authCode = "";
+		java.util.Enumeration<java.lang.String> e = request.getParameterNames();
+		while (e.hasMoreElements()) {
+			String paramName = e.nextElement();
+			if ("code".equals(paramName)) {
+				authCode = request.getParameter(paramName);
+				log.info("authCode:" + authCode);
```

Comment:
Is this sensitive?  Should only enough of the code be logged to make sure it can be identified?